src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1115,6 +1115,26 @@ private static void ExpandTypeName(TypeSyntax type, StringBuilder builder)
                     builder.Append(qualified.DotToken.Text);
                     ExpandTypeName(qualified.Right, builder);
                     break;
+                case SyntaxKind.TupleType:
```

Comment:
So... uh... AFAICT, all this code exists for one purpose.  To help us populate our Navigate-To index with 'full' names for explicitly implemented members.  i.e. if we have:

```
class C
{
    bool IFoo<Bar>.Baz<(int, string)>.Quux() { }
}
```

Instead of just putting "Quux" in the list, we put in ```IFoo<Bar>.Baz<(int, string)>.Quux```.  This seems really unnecessary.  Just putting in "Quux" would be fine as that's going to be what the user is searching for.